### DISCUSSION TITLE
RFC: synchronise to *DAV

### DISCUSSION THREAD

**Discussion by maisim** [OCA/odoo-community.org#89] [Ideas]
(To be completed)

Strangely I cannot find something up to date for either:

- expose calendars via calDAV
- push calendars to a calDAV server

I propose:
First, write a leave_to_caldav module which will write to a caldav server (nextcloud, radicale, ...) (unidirectional)
Each time an event is created or modified:

- Check that the calendar that contains the event exists on the caldav server
- Check that the event exists on the remote calendar
- Create or update the event

Everything will have to be asynchronous so as not to suffer any repercussions in odoo.

Needs:

In odoo:

- Add a dav_UUID field on the models you want to push in caldav (leaves, holidays, ...)
- Add a synchronise_to_dav field?
- Allow users to set remote caldav server (or not)
- Allow users to set their caldav login/password
- Allow users to choose if an item (ie calendar) should be synchronized
- Manage push failures (retries)
- ...

In nextcloud (our use case, as an example):
- create app password for odoo
- ...

In a second step, if useful, we can imagine pushing/synchronizing (bidirectional) files, contacts, ...

- create a generic dav_* module
- use dav_* for leave_to_caldav
- use dav_* for contact_to_cardDAV
- ...

To do this, ensure that the code is sufficiently generic from the start

Your opinions, needs, warnings, tips, ... are welcome

THANKS !

**Comment by brandon50124** [comment]
- - [ ] # ~~****~~

**Comment by brandon50124** [comment]
> (To be completed)
> 
> Strangely I cannot find something up to date for either:
> 
> - expose calendars via calDAV
> - push calendars to a calDAV server
> 
> I propose:
> First, write a leave_to_caldav module which will write to a caldav server (nextcloud, radicale, ...) (unidirectional)
> Each time an event is created or modified:
> 
> - Check that the calendar that contains the event exists on the caldav server
> - Check that the event exists on the remote calendar
> - Create or update the event
> 
> Everything will have to be asynchronous so as not to suffer any repercussions in odoo.
> 
> Needs:
> 
> In odoo:
> 
> - Add a dav_UUID field on the models you want to push in caldav (leaves, holidays, ...)
> - Add a synchronise_to_dav field?
> - Allow users to set remote caldav server (or not)
> - Allow users to set their caldav login/password
> - Allow users to choose if an item (ie calendar) should be synchronized
> - Manage push failures (retries)
> - ...
> 
> In nextcloud (our use case, as an example):
> - create app password for odoo
> - ...
> 
> In a second step, if useful, we can imagine pushing/synchronizing (bidirectional) files, contacts, ...
> 
> - create a generic dav_* module
> - use dav_* for leave_to_caldav
> - use dav_* for contact_to_cardDAV
> - ...
> 
> To do this, ensure that the code is sufficiently generic from the start
> 
> Your opinions, needs, warnings, tips, ... are welcome
> 
> THANKS !

**Comment by bosd** [comment]
Maybe this is of interest to you:
https://github.com/OCA/server-backend/pull/276
https://github.com/OCA/server-backend/issues/254
